dipy/core/gradients.py

```diff
@@ -667,11 +667,14 @@ def reorient_bvecs(gtab, affines, atol=1e-2):
     """
     new_bvecs = gtab.bvecs[~gtab.b0s_mask]
```

Comment:
we might need to add a warning if we detect that the shape is `(n, 4, 4) or (n, 3, 3)`. For a long time, it was like that, this change can break some code. So better to warn them

Reply:
Should I correct the dimension if the shape is wrong and throw a warning that it will be not compatible in future releases?

Reply:
Yes !